guides/quickstart.md

```diff
@@ -0,0 +1,175 @@
+
+`auth.go`
+```go
+type Authentication struct {
+  Username string 
+  Password string `datapolicy:"secret"`
+}
+```
+
+If you do not wish to define your own field tag, you may use the built-in `levee:"source"` tag.
+
+#### Using type descriptions
+
+In the analyzer's configuration, identify the sensitive data:
+
+`analyzer_configuration.yaml`
+```yaml
+Sources:
+  - Package: auth
+    Type: Authentication
+    Field: Password
+```
+
+This method does not require changes to the code.
+
+See [the documentation](../configuration/README.md) for further instructions on how to describe sources.
+
+### Configuring a Sink
+
+In taint propagation lingo, a function that should not be called with a tainted value is called a "Sink".
+In the current example, a Sink is a function that could leak a sensitive value, e.g. by writing it to a log.
+
+Configuring a sink is similar to configuring a source using a type description.
+In the analyzer's configuration, identify the function:
+
+```yaml
+Sinks:
+  - Package: log
+    Method: Printf
+```
+
+See [the documentation](../configuration/README.md) for further instructions on how to describe sinks.
+
+## Running the analyzer
+
+Now that the analyzer knows what our Sources and Sinks are, we can run it to detect the issue in the example code.
+
+### Install the analyzer
+
+Use the following command to install the analyzer:
+
+```shell
+go get github.com/google/go-flow-levee/cmd/levee
+```
+
+### Run the analyzer
+
+The most convenient way to run the analyzer is to use the `go vet` command.
+
+You must provide `go vet` with three pieces of information:
+1. What tool to run (in this case, the levee analyzer)
+2. What configuration the analyzer should use
+3. The list of packages to analyze
+
+Use the following command to run the analyzer:
+```shell
+go vet -vettool=$(which levee) -config=$(realpath analyzer_configuration.yaml) ./...
+```
+
+(This command assumes that it is being run in the `go-flow-levee/guides/quickstart` directory.)
+
+Running on the example code, you should see output similar to the following:
+
+```
+# github.com/google/go-flow-levee/guides/quickstart
+./auth.go:13:13: a source has reached a sink
+ source: ./auth.go:10:19
+```
+
+The analyzer detected that a source could reach a sink, and it produced a helpful report
+indicating the locations of the source and sink in the code.
+
+Let's fix the issue. Do we really need to be logging the `auth` struct? Maybe not:
+
+`auth.go`
+```go
+		log.Printf("unable to make authenticated request: incorrect authentication?\n")
+```
+
+After making this change, the analyzer no longer produces a report. There are many ways to
+address reports, the important thing is to modify the code so that the Source can no longer
+reach the Sink. For example, you may wish to define a "Sanitizer" function that removes the
+password, making the `auth` struct safe to log. See [the documentation](../configuration/README.md)
```

Comment:
`Authentication`

Reply:
Are you referring to `auth`? I think I prefer `auth` here, because that is IMO the most specific label, being the name of the variable in the example. WDYT?

Reply:
It says making the `auth` struct safe to log. But there is no `auth` struct there is an `Authentication` struct.

Reply:
In my understanding, the `auth` variable is a `struct` of type `Authentication`. I see your point, though. The sanitizer would work with values of type `Authentication`.

I've reworded it. WDYT?

Reply:
I am thinking do we really need the line
`For example, you may wish to define ...`